pkg/cmd/util/clientcmd/factory_client_access.go

```diff
@@ -103,7 +103,7 @@ func DefaultClientConfig(flags *pflag.FlagSet) kclientcmd.ClientConfig {
 	cobra.MarkFlagFilename(flags, config.OpenShiftConfigFlagName)
 
 	// set our explicit defaults
-	defaultOverrides := &kclientcmd.ConfigOverrides{ClusterDefaults: kclientcmdapi.Cluster{Server: os.Getenv("KUBERNETES_MASTER")}}
+	defaultOverrides := &kclientcmd.ConfigOverrides{ClusterDefaults: kclientcmd.ClusterDefaults}
```

Comment:
I don't think this is right... I think this is just making the config appear non-empty (server="localhost:8080" instead of ""), which is triggering a different code path